$MirOS: ports/multimedia/mplayer/patches/patch-get_path_c,v 1.1.7.1 2005/03/18 15:50:49 tg Exp $
--- get_path.c.orig	Wed Apr 13 11:45:50 2005
+++ get_path.c	Thu May 19 20:32:06 2005
@@ -24,7 +24,7 @@ char *get_path(char *filename){
 #if defined(__MINGW32__)
     static char *config_dir = "/mplayer";
 #else
-	static char *config_dir = "/.mplayer";
+	static char *config_dir = "/.etc/mplayer";
 #endif
 	int len;
 #ifdef MACOSX_BUNDLE
@@ -48,12 +48,12 @@ char *get_path(char *filename){
 	if (filename == NULL) {
 		if ((buff = (char *) malloc(len)) == NULL)
 			return NULL;
-		sprintf(buff, "%s%s", homedir, config_dir);
+		snprintf(buff, len, "%s%s", homedir, config_dir);
 	} else {
 		len += strlen(filename) + 1;
 		if ((buff = (char *) malloc(len)) == NULL)
 			return NULL;
-		sprintf(buff, "%s%s/%s", homedir, config_dir, filename);
+		snprintf(buff, len, "%s%s/%s", homedir, config_dir, filename);
 	}
 
 #ifdef MACOSX_BUNDLE
